ourhexenv.py

```diff
@@ -118,17 +121,21 @@ def step(self, action):
                 self.terminations = {agent: True for agent in self.agents}
             else:
                 if not self.sparse_flag:
-                    self.rewards = {agent: -1 for agent in self.agents} 
+                    self.rewards = {agent: -1 for agent in self.agents}
+                    self.rewards[self.agent_selection] = -1
+
 
-        
         if self.agent_selection == "player_2":
             # Player 2 has made their first move, make pie rule unusable
             self.is_pie_rule_usable = False
 
         for agent in self.agents:
             self._cumulative_rewards[agent] += self.rewards[agent]
 
-        self.agent_selection = self.agent_selector.next()
+        #Advance to the next agent if the game is not over
+        if not any(self.terminations.values()):
+            self.agent_selection = self.agent_selector.next()
```

Comment:
Is this necessary? Seems redundant and confusing.

Reply:
Not sure what you mean. We do need to include 
`self.agent_selector.next()` 
so that the env can proceed to the next agent.

Reply:
I meant the if condition?